### PR TITLE
field changes for Publishable Positions

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
[TM-5833](https://metaphase.atlassian.net/browse/TM-5833)

1. Rename the following read only fields:
    - [ ] ~Organization → Location/Org~ - we do not currently show location org, just org, so it doesn’t make sense to rename this
    - [x] Status → Publishable Status
    - [ ] ~Language → Language/Training~ - we do not currently show training, just language, so it doesn’t make sense to rename this
    - [x] Tour of Duty → Default TOD
2. Rename the following edit form fields:
    - [x] Override Tour of Duty → Override Position TOD
    - [x] Future Cycle → Proposed Cycle
    - [x] Add Default TOD to Edit form
    - [x] Status → Publishable Status
    - [x] Removed Status from the read-only part of the Edit Form, since that value is editable and is presented in a dropdown
3. Add the following fields to read-only section - added them statically to the edit form, as well to avoid confusion
    - [x] Employee ID
        * This is the Incumbent’s employee ID
    - [x] Employee Status 
        * ex: Effected
        * This is the Incumbent’s status
4. Combine Pay Plan and Grade fields
    - [x] PP/Grade
5. - [x] Added missing static fields to the Edit mode to not have fields disappearing on the user when they flip to edit mode

[TM-5833]: https://metaphase.atlassian.net/browse/TM-5833?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/utilities.js [hunk 1]
```diff
@@ -1123,4 +1123,21 @@ export const joinIfThere = (array, defaultText = 'None Listed', orderMatters = f
...
+  (falsy, falsy)  ->  'None Listed'
+  (falsy, '02')   ->  'None Listed/02'
+  ('FP', falsy)   ->  'FP/None Listed'
+*/
+export const ppGrade = (pp, grade) => {
+  const defaultText = 'None Listed';
+  if (!pp && !grade) {
+    return defaultText;
+  }
+  return `${pp || defaultText}/${grade || defaultText}`;
```
Comment (elizabeth-jimenez) [inline src/utilities.js:1140]: looking at older implementations, i think they dont want the text to have a `/`. will update

Comment (elizabeth-jimenez) [reply src/utilities.js]: done

File: src/utilities.js [hunk 2]
```diff
@@ -1123,4 +1123,21 @@ export const joinIfThere = (array, defaultText = 'None Listed', orderMatters = f
...
+/*
+  Search Tags: pay plan grade, pp/grade, format pay plan, format grade, format pp
+
+  Examples:
+  ('FP', '02')    ->  'FP/02'
+  (falsy, falsy)  ->  'None Listed'
+  (falsy, '02')   ->  'None Listed/02'
+  ('FP', falsy)   ->  'FP/None Listed'
+*/
+export const ppGrade = (pp, grade) => {
```
Comment (elizabeth-jimenez) [inline src/utilities.js:1135]: moving this to the BE in another PR where I try to standardize pp grade

Comment (jlam-mpc) [reply src/utilities.js:1135]: Does this mean you'll be merging the two fields across the board? I think we need to keep them separate, no?

File: src/utilities.js [hunk 3]
```diff
@@ -1123,4 +1123,21 @@ export const joinIfThere = (array, defaultText = 'None Listed', orderMatters = f
...
+  const defaultText = 'None Listed';
```
Comment (jlam-mpc) [inline src/utilities.js:1136]: If you don't move this to the BE, I would import DEFAULT_TEXT and use it in the above line so that "None Listed" isn't manually declared in too many places!

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 4]
```diff
@@ -117,18 +118,16 @@ const PublishablePositionCard = ({
     staticBody: [
       { 'Bureau': data?.bureau || DEFAULT_TEXT },
       { 'Organization': data?.org || DEFAULT_TEXT },
-      { 'Grade': data?.grade || DEFAULT_TEXT },
-      { 'Status': data?.status || DEFAULT_TEXT },
```
Comment (jlam-mpc) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:121]: Good call on removing this in the read only section of the form!

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 5]
```diff
@@ -252,6 +251,18 @@ const PublishablePositionCard = ({
...
 
+  if (PP_FLAG()) {
+    form.staticBody.push(
+      { TED: data?.status || DEFAULT_TEXT },
+      { Incumbent: data?.status || DEFAULT_TEXT },
+      { 'Default TOD': data?.status || DEFAULT_TEXT },
+      { Assignee: data?.status || DEFAULT_TEXT },
+      { 'Post Differential | Danger Pay': data?.status || DEFAULT_TEXT },
+      { 'Employee ID': data?.status || DEFAULT_TEXT },
+      { 'Employee Status': data?.status || DEFAULT_TEXT },
```
Comment (jlam-mpc) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:262]: I'd want to suggest declaring this at the top of the file:

```
const additionalRO = [
      { TED: data?.status || DEFAULT_TEXT },
      { Incumbent: data?.status || DEFAULT_TEXT },
      { 'Default TOD': data?.status || DEFAULT_TEXT },
      { Assignee: data?.status || DEFAULT_TEXT },
      { 'Post Differential | Danger Pay': data?.status || DEFAULT_TEXT },
      { 'Employee ID': data?.status || DEFAULT_TEXT },
      { 'Employee Status': data?.status || DEFAULT_TEXT },
]
```

and then on line 49:

```
    bodySecondary: PP_FLAG() ?
      [
        { 'Bid Cycle': data?.status || DEFAULT_TEXT },
        ...additionalRO,
      ]
      : [],
```

and on line 259:

```
  if (PP_FLAG()) {
    form.staticBody.push(additionalRO);
  }
```

**Review by jlam-mpc — COMMENTED** [review]
Everything looks good- checked your todo list and the AC on the ticket and it looks like everythings been touched on + more was improved. I left one suggestion to reduce redundancy in that giant block of read only lines if you would like to make that change (this only applies to that big block because the same fields and order are retained).

File: src/utilities.js (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/utilities.js:1135]: i'm merging them, but still passing along the other two - tho i dont think we really need them, just playing it safe

File: src/utilities.js (hunk 3 above)
Comment (elizabeth-jimenez) [reply src/utilities.js:1136]: i moved it, it's here https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/964/files

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx (hunk 5 above)
Comment (elizabeth-jimenez) [reply src/Components/PublishablePositionCard/PublishablePositionCard.jsx]: beautiful suggestion, thank you! Done!

**Review by jlam-mpc — APPROVED** [review]
